ui/app/src/components/shared/ResourceCard.tsx

```diff
@@ -1,116 +1,158 @@
-import React, { useState } from 'react';
-import { Resource } from '../../models/resource';
-import { Callout, DefaultPalette, FontWeights, IconButton, mergeStyleSets, PrimaryButton, Stack, Text } from '@fluentui/react';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { ComponentAction, getResourceFromResult, Resource, ResourceUpdate } from '../../models/resource';
+import { Callout, DefaultPalette, FontWeights, IconButton, IContextualMenuItem, IContextualMenuProps, mergeStyleSets, PrimaryButton, ProgressIndicator, Shimmer, Stack, Text } from '@fluentui/react';
 import { Link } from 'react-router-dom';
 import moment from 'moment';
+import { RoleName, WorkspaceRoleName } from '../../models/roleNames';
+import { SecuredByRole } from './SecuredByRole';
+import { ResourceType } from '../../models/resourceType';
+import { ConfirmDisableEnableResource } from './ConfirmDisableEnableResource';
+import { NotificationsContext } from '../../contexts/NotificationsContext';
+import { HttpMethod, useAuthApiCall } from '../../useAuthApiCall';
+import { WorkspaceContext } from '../../contexts/WorkspaceContext';
+import { ConfirmDeleteResource } from './ConfirmDeleteResource';
 
 interface ResourceCardProps {
   resource: Resource,
   itemId: number,
   selectResource: (resource: Resource) => void,
-  contextMenuElement?: JSX.Element
+  onUpdate: (resource: Resource) => void,
+  onDelete: (resource: Resource) => void
 }
 
 export const ResourceCard: React.FunctionComponent<ResourceCardProps> = (props: ResourceCardProps) => {
+  const apiCall = useAuthApiCall();
+  const workspaceCtx = useContext(WorkspaceContext);
+  const [loading, setLoading] = useState(false);
   const [showInfo, setShowInfo] = useState(false);
-
-  const cardStyles: React.CSSProperties = {
-    width: '100%',
-    borderRadius: '2px',
-    border: '1px #ccc solid'
+  const [showDisable, setShowDisable] = useState(false);
+  const [showDelete, setShowDelete] = useState(false);
+  const [componentAction, setComponentAction] = useState(ComponentAction.None);
+
+  const opsReadContext = useContext(NotificationsContext);
+  const opsWriteContext = useRef(useContext(NotificationsContext)); // useRef to avoid re-running a hook on context write
+
+  // set the latest component action
+  useEffect(() => {
+    let updates = opsReadContext.resourceUpdates.filter((r: ResourceUpdate) => { return r.resourceId === props.resource.id });
+    setComponentAction((updates && updates.length > 0) ?
+      updates[updates.length - 1].componentAction :
+      ComponentAction.None);
+  }, [opsReadContext.resourceUpdates, props.resource.id])
+
+  // act on component action changes
+  useEffect(() => {
+    const checkForReload = async () => {
+      if (componentAction === ComponentAction.Reload) {
+        setLoading(true);
+        let result = await apiCall(props.resource.resourcePath, HttpMethod.Get, workspaceCtx.workspaceClientId);
```

Comment:
We might want to think about error handling here if the call fails (i.e. showing something to the user) - feel free to defer to a larger story that tackles error handling in general

Reply:
yea let's revisit error messages later, as we might be able to do something centrally perhaps